src/JustSaying/AwsTools/MessageHandling/SnsTopicByName.cs

```diff
@@ -85,49 +85,49 @@ public override async Task<bool> ExistsAsync()
             return false;
         }
 
-        public async Task<bool> CreateAsync()
+        public async Task CreateAsync()
         {
             try
             {
                 var response = await Client.CreateTopicAsync(new CreateTopicRequest(TopicName))
                     .ConfigureAwait(false);
 
-                if (!string.IsNullOrEmpty(response.TopicArn))
+                if (string.IsNullOrEmpty(response.TopicArn))
                 {
-                    Arn = response.TopicArn;
-                    _log.LogInformation("Created topic '{TopicName}' on ARN '{Arn}'.", TopicName, Arn);
-                    return true;
+                    _log.LogError("Failed to create topic or obtain ARN for topic {TopicName}", TopicName);
```

Comment:
I would not repeat the word `topic` as you are looking for the ARN of the created or existing topic. 
```suggestion
                    _log.LogError("Failed to create or obtain ARN for topic {TopicName}", TopicName);
```

Reply:
Done 👍